src/SimpleCacheTest.php

```diff
@@ -0,0 +1,589 @@
+            $this->assertNull($r);
+        }
+
+        $this->cache->set('key3', 'value');
+        $result = $this->cache->getMultiple(['key2', 'key3', 'key4'], 'foo');
+        foreach ($result as $key => $r) {
+            if ($key === 'key3') {
+                $this->assertEquals('value', $r);
+            }
+
+            $this->assertEquals('foo', $r);
+        }
+    }
+
+    public function testGetMultipleWithGenerator()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $gen = function () {
+            yield 'key0';
+            yield 'key1';
+        };
+
+        $this->cache->set('key0', 'value0');
+        $this->cache->getMultiple($gen);
+        $this->assertEquals('value0', $this->cache->get('key0'));
+        $this->assertNull($this->cache->get('key1'));
+    }
+
+    public function testDeleteMultiple()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $this->assertTrue($this->cache->deleteMultiple([]), 'Deleting a empty array should return true');
+        $this->assertTrue($this->cache->deleteMultiple(['key']), 'Deleting a value that does not exist should return true');
+
+        $this->cache->set('key0', 'value0');
+        $this->cache->set('key1', 'value1');
+        $this->assertTrue($this->cache->deleteMultiple(['key0', 'key1']), 'Delete must return true on success');
+        $this->assertNull($this->cache->get('key0'), 'Values must be deleted on deleteMultiple()');
+        $this->assertNull($this->cache->get('key1'), 'Values must be deleted on deleteMultiple()');
+    }
+
+    public function testDeleteMultipleGenerator()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $gen = function () {
+            yield 'key0';
+            yield 'key1';
+        };
+        $this->cache->set('key0', 'value0');
+        $this->assertTrue($this->cache->deleteMultiple($gen), 'Deleting a generator should return true');
+
+        $this->assertNull($this->cache->get('key0'), 'Values must be deleted on deleteMultiple()');
+        $this->assertNull($this->cache->get('key1'), 'Values must be deleted on deleteMultiple()');
+    }
+    public function testHas()
+    {
+        if (isset($this->skippedTests[__FUNCTION__])) {
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $this->assertFalse($this->cache->has('key0'));
+        $this->cache->set('key0', 'value0');
+        $this->assertTrue($this->cache->has('key0'));
+
+        $this->cache->set('key1', null);
+        $this->assertFalse($this->cache->has('key1'), 'A value of null is considered as has=false');
```

Comment:
From 1.2 in the standard: 

> A cache miss will return null and therefore detecting if one stored null is not possible. This is the main deviation from PSR-6's assumptions.

Since we cant detect if one stored null we can not return true on `has()`. That is my interpretation.